### PR TITLE
DDF-2827 switched the default authentication mechanism to the IdP

### PR DESCRIPTION
**Pull request by stustison**
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@tbatie @ahoffer @brendan-hofmann @coyotesqrl 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Security](https://github.com/orgs/codice/teams/security)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@stustison
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: platform/security/idp/security-idp-server/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <OCD description="IdP Server" name="IdP Server" id="org.codice.ddf.security.idp.server.IdpEndpoint">
 
         <AD name="SP Metadata" id="spMetadata" required="true" type="String"
-            default="" cardinality="100"
+            default="https://${org.codice.ddf.system.hostname}:${org.codice.ddf.system.httpsPort}/services/saml/sso/metadata" cardinality="100"
```
Comment (coyotesqrl) [inline platform/security/idp/security-idp-server/src/main/resources/OSGI-INF/metatype/metatype.xml]: Nit: `cardinality` should be moved to the next line.

File: platform/security/policy/security-policy-context/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@
         <AD description="List of authentication types required for each context. List of default valid authentication types are: SAML, BASIC, PKI, GUEST. Example: /context=AUTH1|AUTH2|AUTH3"
```
Comment (coyotesqrl) [inline platform/security/policy/security-policy-context/src/main/resources/OSGI-INF/metatype/metatype.xml]: ❗️ Description here should be updated to include `IDP` as a valid authentication type.

**Comment by pklinef** [comment]
Do we need to update any itests to use the IdP as well?

**Comment by stustison** [comment]
Yeah, all the itests are broken and I'm currently fixing them. I just wanted to get the actual change up here to be reviewed without needing to look at 15 itests classes that don't matter.

File: platform/security/policy/security-policy-context/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 3]
```diff
@@ -29,10 +29,10 @@
             type="String"
             default="/=karaf"/>
 
-        <AD description="List of authentication types required for each context. List of default valid authentication types are: SAML, BASIC, PKI, GUEST. Example: /context=AUTH1|AUTH2|AUTH3"
+        <AD description="List of authentication types required for each context. List of default valid authentication types are: IDP, SAML, BASIC, PKI, GUEST. Example: /context=AUTH1|AUTH2|AUTH3"
             name="Authentication Types" id="authenticationTypes" required="true" cardinality="100"
             type="String"
-            default="/=SAML|GUEST,/admin=SAML|basic,/system=basic,/solr=SAML|PKI|basic,/sources=SAML|basic,/security-config=SAML|basic"/>
+            default="/=IDP,/solr=SAML|PKI|BASIC"/>
```
Comment (tbatie) [inline platform/security/policy/security-policy-context/src/main/resources/OSGI-INF/metatype/metatype.xml]: The docs need to be updated to reflect these new default values.

**Comment by stustison** [comment]
The itest changes are up, if anyone cares about those.

File: platform/security/policy/security-policy-context/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 3 above)
Comment (stustison) [inline platform/security/policy/security-policy-context/src/main/resources/OSGI-INF/metatype/metatype.xml]: I looked in the docs and there aren't any places to update. We have example configurations in the documentation, but those should remain.

**Pull request merged**